android/tests_backend/widgets/base.py

```diff
@@ -39,11 +77,13 @@ def hidden(self):
 
     @property
     def width(self):
-        return self.native.getWidth()
+        # Return the value in SP
+        return self.native.getWidth() / self.scale_factor
 
     @property
     def height(self):
-        return self.native.getHeight()
+        # Return the value in SP
+        return self.native.getHeight() / self.scale_factor
```

Comment:
This is actually calculating DP, not SP. The comments are wrong, but the implementation is correct – sizes of things other than fonts should be measured in DP, so they produce a consistent physical size relative to a fingertip.